File: 求根节点到叶节点数字之和.cpp
class Solution {
public:
    int ans = 0;

    void dfs(TreeNode* root, int sum) {
        if (root == NULL) {
            return;
        }
        sum = sum * 10 + root->val;
        if (root->left == NULL and root->right == NULL) {
            ans += sum;
            return;
        }
        if (root->left != NULL) {
            dfs(root->left, sum);
        }
        if (root->right != NULL) {
            dfs(root->right, sum);
        }
    }
    int sumNumbers(TreeNode* root) {
        if (root == NULL) {
            return 0;
        }
        dfs(root, 0);
        return ans;
    }
};

File: 求根节点到叶节点数字之和.py
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        ans = 0
        def dfs(node, x):
            if node is None:
                return
            x = x * 10 + node.val
            if node.left is None and node.right is None:
                nonlocal ans
                ans += x
                return
            if node.left:
                dfs(node.left, x)
            if node.right:
                dfs(node.right, x)

        dfs(root, 0)
        return ans
